File: c/resistor-color/resistor_color.c
#include "resistor_color.h"
#include <stdlib.h>


int color_code(resistor_band_t color_code)
{
    return color_code;
}

resistor_band_t* colors()
{

    resistor_band_t* ret = malloc(10 * sizeof(resistor_band_t));


    if (ret == NULL) {
        return NULL; 
    }

    // Fill in the array with the enum values
    ret[0] = BLACK;
    ret[1] = BROWN;
    ret[2] = RED;
    ret[3] = ORANGE;
    ret[4] = YELLOW;
    ret[5] = GREEN;
    ret[6] = BLUE;
    ret[7] = VIOLET;
    ret[8] = GREY;
    ret[9] = WHITE;

    return ret;
}

File: solutions/clojure/resistor-color/1/src/resistor_color.clj
(ns resistor-color)

(def colors
  ;; definition value
  ["black" "brown" "red" "orange" "yellow" "green" "blue" "violet" "grey" "white"])

(defn color-code
  "Returns the numerical value associated with the given color."
  [color]
  ;; function body
  (cond (= color "black") 0
        (= color "brown") 1
        (= color "red") 2
        (= color "orange") 3
        (= color "yellow") 4
        (= color "green") 5
        (= color "blue") 6
        (= color "violet") 7
        (= color "grey") 8
        (= color "white") 9))
